### ISSUE TITLE
Add support for raw POST like curl's --data option

### ISSUE THREAD

**Issue by birtles** [nathan-osman/REST-Easy#64]
Specifically, I was trying to use this with W3C's [Echidna publication system](https://github.com/w3c/echidna/wiki/How-to-use-Echidna#url-or-manifest) but ran it to trouble since it complained that it doesn't support form data.

**Comment by mirkoperillo** [comment]
Curl documentation says that 

```
-d, --data <data>
              (HTTP) Sends the specified data in a POST request  to  the  HTTP
              server,  in  the  same  way  that a browser does when a user has
              filled in an HTML form and presses the submit button. This  will
              cause curl to pass the data to the server using the content-type
              application/x-www-form-urlencoded.
```

So doing a POST with Form body and MIME type application/x-www-form-urlencoded should work

**Comment by birtles** [comment]
Yes, I read that too and expected the same. Unfortunately in practice it is not the same. `curl --data` works but form with `application/x-www-form-urlencoded` in REST easy does not.

**Comment by mirkoperillo** [comment]
Do you have error or warning in javascript console or some message / http code error in response ?

This can be useful to understand where is the issue

**Comment by birtles** [comment]
Sorry for the delay, you can reproduce simply by doing a POST to https://labs.w3.org/echidna/api/request
The result returned is:
> `{"status":"Form Content-Type not supported"}`

I get a 501 Not Implemented return code with the following headers:

Name |Value
--- | ---
X-Powered-By | Express
Content-Type | application/json; charset=utf-8
Content-Length | 44
Etag | W/"2c-De5+vJK+2lj7uobm3ktF6w"
Vary | Accept-Encoding
Date | Thu, 15 Sep 2016 05:29:28 GMT
Strict-Transport-Security | max-age=15552015; preload
Public-Key-Pins | pin-sha256=.....

**Comment by mirkoperillo** [comment]
I tried using the master branch version and it seems to work.

I tried a POST request with Form body, application/x-www-form-urlencoded MIME, given fake url, decision and token params and the service returns 202 http code and a Uuid

**Comment by mirkoperillo** [comment]
This issue is silent for a month, maybe it can be closed.

If @birtles has news about he can always reopen it

**Comment by birtles** [comment]
Thanks. Sorry, I haven't had time to try install a trunk version to test but I trust it works!
